packages/core-data/src/private-selectors.ts

```diff
@@ -240,38 +240,13 @@ export const getTemplateId = createRegistrySelector(
 		// First see if the post/page has an assigned template and fetch it.
 		const currentTemplateSlug = editedEntity.template;
 		if ( currentTemplateSlug ) {
-			const userTemplates = select( STORE_NAME ).getEntityRecords(
-				'postType',
-				'wp_template',
-				{ per_page: -1 }
-			);
-			if ( ! userTemplates ) {
-				return;
-			}
-			const userTemplateWithSlug = userTemplates.find(
-				( { slug } ) => slug === currentTemplateSlug
-			);
-
-			if ( userTemplateWithSlug ) {
-				return userTemplateWithSlug.id;
-			}
-
-			const registeredTemplates = select( STORE_NAME ).getEntityRecords(
-				'postType',
-				'wp_registered_template',
-				{ per_page: -1 }
-			);
-
-			if ( ! registeredTemplates ) {
-				return;
-			}
-
-			const registeredTemplateWithSlug = registeredTemplates.find(
-				( { slug } ) => slug === currentTemplateSlug
-			);
-
-			if ( registeredTemplateWithSlug ) {
-				return registeredTemplateWithSlug.id;
+			const currentTemplate = select( STORE_NAME )
+				.getEntityRecords( 'postType', 'wp_template', {
+					per_page: -1,
+				} )
+				?.find( ( { slug } ) => slug === currentTemplateSlug );
+			if ( currentTemplate ) {
+				return currentTemplate.id;
```

Comment:
This should have been done in #72661 because it specifically handles the `template` property on a post object.